rules/java/third_parties/bigquery.yml

```diff
@@ -0,0 +1,59 @@
+imports:
+  - java_shared_lang_datatype
+  - java_shared_lang_instance
+patterns:
+  - pattern: $<BUILDER>.addRow($<HASH_MAP>);
+    filters:
+      - variable: BUILDER
+        detection: java_third_parties_bigquery_builder
+      - variable: HASH_MAP
+        detection: java_shared_lang_instance
+        scope: cursor
+        filters:
+          - variable: JAVA_SHARED_LANG_INSTANCE_TYPE
+            regex: \A(java\.util\.)?HashMap\<\>\z
+trigger:
+  required_detection: java_third_parties_bigquery_row_with_data
```

Comment:
We need to match on a HashMap that has contains sensitive data. Not sure how to catch a case like the below, for example, while excluding plain old `new HashMap<>();`

```java
Map<String, String> rowContent = new HashMap<>();
rowContent.put("name", user.name);
```

Reply:
@cfabianski could we discuss this - when you are free :)

Reply:
Similar case with Segment - https://github.com/Bearer/bearer-rules/pull/267

Reply:
This isn't possible at the moment. I think what you have is the best approach for now.